diff --git a/include/Makefile.am b/include/Makefile.am
index 6ed7cda..622e68a 100644
--- a/include/Makefile.am
+++ b/include/Makefile.am
@@ -15,8 +15,7 @@ noinst_HEADERS += util/*.h
 noinst_HEADERS += vm/*.h
 
 install-data-local:
-	destname=spl-$(SPL_META_VERSION)/$(LINUX_VERSION); \
-	instdest=$(DESTDIR)/${prefix}/src/$$destname; \
+	instdest=$(DESTDIR)/${includedir}/spl; \
 	instfiles=`find . -name '*.h'`; \
         for instfile in $$instfiles; do \
 		$(INSTALL) -D $$instfile $$instdest/$$instfile; \
diff --git a/Makefile.am b/Makefile.am
index 8c4fed9..af36071 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -26,8 +26,7 @@ distclean-local::
 
 if CONFIG_KERNEL
 install-data-local:
-	destname=spl-$(SPL_META_VERSION)/$(LINUX_VERSION); \
-	instdest=$(DESTDIR)/${prefix}/src/$$destname; \
+	instdest=$(DESTDIR)/${includedir}/spl; \
 	echo "${SPL_META_VERSION}" >$$instdest/spl.release; \
 	for instfile in $(noinst_HEADERS) module/$(LINUX_SYMBOLS); do \
 		$(INSTALL) -D $$instfile $$instdest/$$instfile; \
