From ff4b6d270544a94cb46e8f8e1a961776582d5be1 Mon Sep 17 00:00:00 2001
From: Henrik Sandklef <hesa@gnu.org>
Date: Sat, 9 May 2009 03:21:32 +0200
Subject: [PATCH] Fix doc compile error

---
 doc/xnee.texi |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/doc/xnee.texi b/doc/xnee.texi
index 03290fa..c209a51 100644
--- a/doc/xnee.texi
+++ b/doc/xnee.texi
@@ -52,8 +52,8 @@
 @end titlepage
 
 
-@ifinfo
 @node Top, Summary, (dir), (dir)
+@ifinfo
 @top 
 @end ifinfo
 
-- 
1.6.3.rc0.1.gf800.dirty

