From aa39985ea94ed09c27e56da68652e291f1469779 Mon Sep 17 00:00:00 2001
From: Sebastian Pipping <sebastian@pipping.org>
Date: Mon, 8 Apr 2013 21:48:23 +0200
Subject: [PATCH] Make nm-applet 0.9.4.1 systray icon as small as requested
 (v1)

---
 src/applet.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/applet.c b/src/applet.c
index be4e516..0c69ecf 100644
--- a/src/applet.c
+++ b/src/applet.c
@@ -3033,7 +3033,7 @@ nma_icon_check_and_load (const char *name, GdkPixbuf **icon, NMApplet *applet)
 	/* Try to load the icon; if the load fails, log the problem, and set
 	 * the icon to the fallback icon if requested.
 	 */
-	*icon = gtk_icon_theme_load_icon (applet->icon_theme, name, applet->icon_size, 0, &error);
+	*icon = gtk_icon_theme_load_icon (applet->icon_theme, name, applet->icon_size, GTK_ICON_LOOKUP_FORCE_SIZE, &error);
 	if (!*icon) {
 		g_warning ("Icon %s missing: (%d) %s",
 		           name,
@@ -3139,7 +3139,7 @@ status_icon_size_changed_cb (GtkStatusIcon *icon,
 	/* icon_size may be 0 if for example the panel hasn't given us any space
 	 * yet.  We'll get resized later, but for now just load the 16x16 icons.
 	 */
-	applet->icon_size = MAX (16, size);
+	applet->icon_size = MAX (1, size);
 
 	nma_icons_reload (applet);
 
-- 
1.8.1.5

