diff --git a/liblicense-gnome-0.4/work/liblicense-gnome-0.4/src/Makefile.am b/liblicense-gnome-0.4/work/liblicense-gnome-0.4/src/Makefile.am
index 7c88584..bedbfd4 100644
--- a/liblicense-gnome-0.4/work/liblicense-gnome-0.4/src/Makefile.am
+++ b/liblicense-gnome-0.4/work/liblicense-gnome-0.4/src/Makefile.am
@@ -30,7 +30,7 @@ emblems_DATA=emblem-cc.svg \
 
 #force an update of the icon cache
 install-data-hook:
-	touch $(datadir)/icons/hicolor
+	touch $(DESTDIR)$(datadir)/icons/hicolor
 
 if HAVE_NP
 pynautilusdir = $(libdir)/nautilus/extensions-1.0/python
