--- Makefile
+++ Makefile
@@ -12,13 +12,13 @@ ELKDIR  = /usr/elk
 
 ###  Additional libraries.  You may want to insert the output of the
 ###  shell-script $(ELKDIR)/lib/ldflags here.
-LIBS    = -lm
+LIBS    = -lm -lelk
 
 ###  The makedepend program (it's usually installed with the X11 binaries).
 MAKEDEP = makedepend
 
 ###  The directory under which you will install the Scheme files.
-DIR = /usr/local/lib/unroff
+DIR = /usr/share/unroff/
 
 ###  The default output format.
 FORMAT = html
@@ -30,8 +31,7 @@ FORMAT = html
 ###  ------------------------------------------------------------------------- 
 
 SHELL   = /bin/sh
-INCLUDE = -I$(ELKDIR)/include
-ELK     = $(ELKDIR)/lib/module.o
+INCLUDE = -I/usr/include/elk
 DEFS    = -DDEFAULT_DIR=\"$(DIR)\" -DDEFAULT_FORMAT=\"$(FORMAT)\"
 CTAGS   = ctags -t -w
 
@@ -67,8 +67,7 @@ OBJECTS	= \
 	stream.o\
 	subst.o\
 	table.o\
-	unroff.o\
-	$(ELK)
+	unroff.o
 
 ALL	= unroff
 
--- error.c
+++ error.c
@@ -39,8 +39,9 @@ void fatal_error(char *fmt, ...) {
 
 static char *strerr(void) {
     extern int sys_nerr;
+/*
     extern char *sys_errlist[];
-
+*/
     return errno > 0 && errno < sys_nerr ?
 	sys_errlist[errno] : "unknown error";
 }
--- error.h
+++ error.h
@@ -13,8 +13,8 @@ void write_error(char *) NORETURN;
 void open_error(char *) NORETURN;
 
 /* Redefined to add NORETURN
- */
 extern Primitive_Error() NORETURN;
+*/
 
 char *printable_string(char *, int);
 char *printable_char(char);
