diff --git a/src/nautilus.defs b/src/nautilus.defs
index 439b8f9..65949d3 100644
--- a/src/nautilus.defs
+++ b/src/nautilus.defs
@@ -167,12 +167,6 @@
   (return-type "gboolean")
 )
 
-(define-method get_vfs_file_info
-  (of-object "NautilusFileInfo")
-  (c-name "nautilus_file_info_get_vfs_file_info")
-  (return-type "GnomeVFSFileInfo*")
-)
-
 (define-method add_emblem
   (of-object "NautilusFileInfo")
   (c-name "nautilus_file_info_add_emblem")
diff --git a/src/nautilus.override b/src/nautilus.override
index 3fa3d72..9e1a706 100644
--- a/src/nautilus.override
+++ b/src/nautilus.override
@@ -41,17 +41,3 @@ import gtk.Window as PyGtkWindow_Type
 ignore-glob
   *_get_type
 %%
-override nautilus_file_info_get_vfs_file_info noargs
-static PyObject *
-_wrap_nautilus_file_info_get_vfs_file_info(PyGObject *self)
-{
-    GnomeVFSFileInfo *ret;
-    
-    ret = nautilus_file_info_get_vfs_file_info(NAUTILUS_FILE_INFO(self->obj));
-    if (ret) {
-        return pygnome_vfs_file_info_new(ret);
-    }
-    
-    Py_INCREF(Py_None);
-    return Py_None;
-}
