From 2a10c7536146f223bd99d2d756439999ef35cbb9 Mon Sep 17 00:00:00 2001
From: Sebastian Pipping <sebastian@pipping.org>
Date: Wed, 14 Jul 2010 00:06:16 +0200
Subject: [PATCH] Fix check for files_fttable() assuming presence of linux/fdtable.h

---
 config/spl-build.m4 |    4 +---
 1 files changed, 1 insertions(+), 3 deletions(-)

diff --git a/config/spl-build.m4 b/config/spl-build.m4
index a0fdb3e..6483f52 100644
--- a/config/spl-build.m4
+++ b/config/spl-build.m4
@@ -793,11 +793,9 @@ AC_DEFUN([SPL_AC_FILES_FDTABLE], [
 	SPL_LINUX_TRY_COMPILE([
 		#include <linux/sched.h>
 		#include <linux/file.h>
-		#ifdef HAVE_FDTABLE_HEADER
 		#include <linux/fdtable.h>
-		#endif
 	],[
-		struct files_struct *files = current->files;
+		struct files_struct *files = NULL;
 		struct fdtable *fdt = files_fdtable(files);
 	],[
 		AC_MSG_RESULT(yes)
-- 
1.7.0.1.61.gdc05d.dirty

