From c8ddfaf21ba127bf7dc09bedf3f0634940ea5cdd Mon Sep 17 00:00:00 2001
From: Sebastian Pipping <sebastian@pipping.org>
Date: Sat, 26 Dec 2009 00:17:12 +0100
Subject: [PATCH 3/4] Fix boost compile flags

---
 hox_Client/Makefile |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/hox_Client/Makefile b/hox_Client/Makefile
index cc3379f..cd3d62f 100644
--- a/hox_Client/Makefile
+++ b/hox_Client/Makefile
@@ -28,8 +28,8 @@ DEBUGFLAGS  =
 #DEBUGFLAGS  = -g
 
 #CXXFLAGS = $(WX_CXXFLAGS) -I../lib/boost_1_41_0 -I../lib/asio-1.4.1/include
-CXXFLAGS = $(WX_CXXFLAGS)
-LDFLAGS     = -pthread
+CXXFLAGS = $(WX_CXXFLAGS) -I/usr/include/boost
+LDFLAGS     = -pthread -lboost_thread -lboost_system
 
 # The main source
 MAIN_SRC := \
-- 
1.6.5.6

